android/src/main/kotlin/com/mapbox/maps/mapbox_maps/MapboxMapController.kt

```diff
@@ -242,50 +238,11 @@ class MapboxMapController(
-        val arguments: HashMap<String, Any> = call.arguments as? HashMap<String, Any> ?: return
-        val interactionList = arguments["interaction"] as? List<Any?> ?: return
-        val interaction = _InteractionPigeon.fromList(interactionList)
-        val featuresetDescriptor = com.mapbox.maps.mapbox_maps.pigeons.FeaturesetDescriptor.fromList(interaction.featuresetDescriptor)
-        val interactionType = _InteractionType.valueOf(interaction.interactionType)
-        val stopPropagation = interaction.stopPropagation
-        val id = interaction.identifier.toLong()
-        val filter = interaction.filter.toValue()
-        val radius = interaction.radius
-
-        featuresetDescriptor.featuresetId?.let {
-          when (interactionType) {
-            _InteractionType.TAP -> mapboxMap?.addInteraction(
-              ClickInteraction.featureset(id = it, importId = featuresetDescriptor.importId, filter = filter, radius = radius) { featuresetFeature, context ->
-                listener.onInteraction(context.toFLTMapContentGestureContext(), featuresetFeature.toFLTFeaturesetFeature(), id) { _ -> }
-                return@featureset stopPropagation
-              }
-            )
-            _InteractionType.LONG_TAP -> mapboxMap?.addInteraction(
-              LongClickInteraction.featureset(id = it, importId = featuresetDescriptor.importId, filter = filter, radius = radius) { featuresetFeature, context ->
-                listener.onInteraction(context.toFLTMapContentGestureContext(), featuresetFeature.toFLTFeaturesetFeature(), id) { _ -> }
-                return@featureset stopPropagation
-              }
-            )
-            null -> return
-          }
-        } ?: featuresetDescriptor.layerId?.let {
-          when (interactionType) {
-            _InteractionType.TAP -> mapboxMap?.addInteraction(
-              ClickInteraction.layer(id = it, filter = filter, radius = radius) { featuresetFeature, context ->
-                listener.onInteraction(context.toFLTMapContentGestureContext(), featuresetFeature.toFLTFeaturesetFeature(), id) { _ -> }
-                return@layer stopPropagation
-              }
-            )
-            _InteractionType.LONG_TAP -> mapboxMap?.addInteraction(
-              LongClickInteraction.layer(id = it, filter = filter, radius = radius) { featuresetFeature, context ->
-                listener.onInteraction(context.toFLTMapContentGestureContext(), featuresetFeature.toFLTFeaturesetFeature(), id) { _ -> }
-                return@layer stopPropagation
-              }
-            )
-            null -> return
-          }
-        }
+        interactionsController.addInteraction(messenger, channelSuffix, call)
+        result.success(null)
+      }
+      "interactions#remove_interaction" -> {
+        interactionsController.removeInteraction(call)
         result.success(null)
       }
       "platform#releaseMethodChannels" -> {
```

Comment:
I don't need to do anything with `dispose()` here, right?

Reply:
Unless you need to perform some cleanup upon map disposal

Reply:
I couldn't think of anything -- no 👍